### PR TITLE
[core] Drop LayerType

### PR DESCRIPTION
**Pull request by pozdnyakov**
Drop `LayerType` and its remaining usages.
The generic code should be layer type agnostic.

### CODE REVIEW

File: src/mbgl/renderer/buckets/circle_bucket.hpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ class BucketParameters;
 class CircleBucket : public Bucket {
 public:
     CircleBucket(const BucketParameters&, const std::vector<const RenderLayer*>&);
+    ~CircleBucket() override;
```
Comment (alexshalamov) [inline src/mbgl/renderer/buckets/circle_bucket.hpp]: nit: declare as `final`? Also, for the rest of buckets.

Comment (pozdnyakov) [reply src/mbgl/renderer/buckets/circle_bucket.hpp]: :+1: however, making the whole class `final` makes it more readable

File: src/mbgl/renderer/bucket.hpp [hunk 2]
```diff
@@ -64,9 +47,14 @@ class Bucket : private util::noncopyable {
     bool needsUpload() const {
         return hasData() && !uploaded;
     }
+    // Returns true if this bucket fits the given layer; returns false otherwise.
+    // Implementations of this class check at least that this bucket has
+    // the same layer type with the given layer, but extra checks are also
+    // possible.
+    virtual bool fitsLayer(const style::Layer::Impl&) const { return true; }
```
Comment (alexshalamov) [inline src/mbgl/renderer/bucket.hpp]: supports / accepts? Layer is parameter, `Bucket.supports(Layer)` looks bit better, imo.

Comment (pozdnyakov) [reply src/mbgl/renderer/bucket.hpp]: Indeed, thanks!

**Pull request merged**